frontend/app/tests/unit/specs/data/subscript-number.spec.ts

```diff
@@ -0,0 +1,196 @@
+        stubs: {
+          CopyTooltip: { template: '<div><slot /></div>' },
+          RuiTooltip: { template: '<div><slot /></div>' },
+        },
+      },
+    });
+
+    const subscriptElement = wrapper.find('[data-cy="amount-display-subscript"]');
+    expect(subscriptElement.exists()).toBe(true);
+    expect(subscriptElement.text()).toBe('7');
+  });
+
+  it('shows correct subscript count for very small numbers', () => {
+    const generalStore = useGeneralSettingsStore();
+    generalStore.update({
+      ...generalStore.settings,
+      uiFloatingPrecision: 10,
+    });
+
+    const wrapper = mount(AmountDisplay, {
+      props: {
+        value: bigNumberify('0.000000000123'),
+        noScramble: true,
+        noTruncate: true,
+        fiatCurrency: null,
+        showCurrency: 'none',
+        forceCurrency: true,
+      },
+      global: {
+        plugins: [pinia],
+        stubs: {
+          CopyTooltip: { template: '<div><slot /></div>' },
+          RuiTooltip: { template: '<div><slot /></div>' },
+        },
+      },
+    });
+
+    const subscriptElement = wrapper.find('[data-cy="amount-display-subscript"]');
+    expect(subscriptElement.exists()).toBe(true);
+    expect(subscriptElement.text()).toBe('9');
+  });
+
+  it('handles regular numbers without subscript', () => {
+    const wrapper = mount(AmountDisplay, {
+      props: {
+        value: bigNumberify('1.23'),
+        noScramble: true,
+        noTruncate: true,
+        fiatCurrency: null,
+        showCurrency: 'none',
+        forceCurrency: true,
+      },
+      global: {
+        plugins: [pinia],
+        stubs: {
+          CopyTooltip: { template: '<div><slot /></div>' },
+          RuiTooltip: { template: '<div><slot /></div>' },
+        },
+      },
+    });
+
+    const subscriptElement = wrapper.find('[data-cy="amount-display-subscript"]');
+    expect(subscriptElement.exists()).toBe(false);
+  });
+
+  it('respects disabled subscript setting', () => {
+    const frontendStore = useFrontendSettingsStore();
+    frontendStore.update({
+      ...frontendStore.settings,
+      subscriptDecimals: false,
+    });
+
+    const wrapper = mount(AmountDisplay, {
+      props: {
+        value: bigNumberify('0.0000000815'),
+        noScramble: true,
+        noTruncate: true,
+        fiatCurrency: null,
+        showCurrency: 'none',
+        forceCurrency: true,
+      },
+      global: {
+        plugins: [pinia],
+        stubs: {
+          CopyTooltip: { template: '<div><slot /></div>' },
+          RuiTooltip: { template: '<div><slot /></div>' },
+        },
+      },
+    });
+
+    const subscriptElement = wrapper.find('[data-cy="amount-display-subscript"]');
+    expect(subscriptElement.exists()).toBe(false);
+  });
+
+  it('shows no subscript for number with only 1 leading zero', () => {
+    const generalStore = useGeneralSettingsStore();
+    generalStore.update({
+      ...generalStore.settings,
+      uiFloatingPrecision: 10,
+    });
```

Comment:
the ones that have the same initialization of the floatingPrecision could be grouped with a `describe` and their own `beforeEach` just to avoid repeating common setup.